include/eve/detail/function/simd/common/swizzle_helpers.hpp

```diff
@@ -0,0 +1,78 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2018-2021 Joel FALCOU
+  Copyright 2018-2021 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/meta.hpp>
+#include <eve/detail/abi.hpp>
+#include <eve/detail/function/bit_cast.hpp>
+#include <eve/pattern.hpp>
+
+namespace eve::detail
+{
+  //----------------------------------------------------------------------------------------------
+  // Handle Zeroing
+  template<typename Pack, typename Pattern>
+  EVE_FORCEINLINE auto process_zeros( Pack const& v, Pattern p ) noexcept
+  {
+    constexpr auto sz = cardinal_v<Pack>;
+    using type = typename Pack::value_type;
+
+    if constexpr( p.has_zeros(sz) )
+    {
+      // Turn pattern into 0/~0 then mask
+      auto const impl = [=](auto... i)
+                        {
+                          using i_t = as_integer_t<type,unsigned>;
+                          constexpr i_t nz = ~i_t(0), zz = i_t(0);
+                          return Pack(bit_cast((p(i,sz)==-1 ? zz : nz),as_<type>())...);
+                        };
+
+      return v & apply<sz>(impl);
+    }
+    else
+    {
+      return v;
+    }
+  }
```

Comment:
I tend to think this should be another operation.
And then the compiler should return:

```
operations<bound<broadcast, args>, bound<replace_zeroes, >>
```

Or smth like that

Reply:
I like the idea it encodes the fact we need to process zeros into the type but I wonder if this si not a bit overkill.

Reply:
Also, process_zeros only use is non-complete swizzle arch. SSSE3 doesn't need it, neither ARM or PPC, so embedding it in the operation is maybe pessimistic.